infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironment.java

```diff
@@ -80,18 +95,35 @@ protected KubernetesEnvironment(
     super(internalEnvironment);
     setType(TYPE);
     this.pods = pods;
+    this.deployments = deployments;
     this.services = services;
     this.ingresses = ingresses;
     this.persistentVolumeClaims = persistentVolumeClaims;
     this.secrets = secrets;
     this.configMaps = configMaps;
+    this.podData = new HashMap<>();
+    pods.entrySet()
+        .forEach(
+            e -> {
+              Pod pod = e.getValue();
+              podData.put(e.getKey(), new PodData(pod.getSpec(), pod.getMetadata()));
+            });
+    deployments
+        .entrySet()
+        .forEach(
+            e -> {
+              PodTemplateSpec podTemplate = e.getValue().getSpec().getTemplate();
+              podData.put(
+                  e.getKey(), new PodData(podTemplate.getSpec(), podTemplate.getMetadata()));
```

Comment:
@amisevsk During resolving conflicts with my PR I discovered that
<pre>
PodData {
  "test-deployments": PodData{
                                metadata: 
                                    name: "deployment-pod"
                                    ....
                                spec: ....
                              }
</pre>
It is done by purpose? Would not be better to sync pod name and map key? If it would be better - I can do it in my PR.

Reply:
The corresponding unit test are here https://github.com/eclipse/che/pull/12335/files#diff-7b6478093925317e7c597de94e907f45R258

Reply:
I'm actually not sure which is better -- I don't think keys are really used in a very direct way. IIRC we pretty much always just iterate through the values in the map.

The issues with syncing the `PodData` key with pod name for deployments are
1. We would also need to modify the key for the deployments map, as otherwise using the same key between the two wouldn't work. Also, either the PodData key doesn't match the resource name, or the deployments map key doesn't match the resource name.
2. The actual pod name in a deployment is irrelevant, as pod name is generated from deployment name. This could be confusing in debugging or testing, as this value is basically ignored.

Personally, I'm open to either choice so long as every key in PodData matches some key in the deployments or pods map. I lean slightly towards keeping it the way it is, as this keeps the key in sync with the deployment name, but I'm not firm either way.

One thing that I haven't tested yet but would be interested to know, is if pod name in deployments is even necessary. This could be an edge case for how PodData is handled. I can look into it once I get back to working on this bit of code.